flocker/node/agents/test/test_blockdevice.py

```diff
@@ -193,14 +207,26 @@ def create_blockdevicedeployer(
 def detach_destroy_volumes(api):
     """
     Detach and destroy all volumes known to this API.
+    If we failed to detach a volume for any reason,
```

Comment:
In addition to failures happening as exceptions (not yet seen in test runs), there could be yet-to-be-reasoned-out failures as well:
Some of the tests run in a pull request experienced the following set of actions:

1) Detach volume - success
2) Destroy volume: Failed because volume was ``in-use`` state.

Will modify the loop to try-except that logs and retries. Thanks!